docs/website/docs/v1.x/dsl/middleware.md

```diff
@@ -3,4 +3,422 @@ sidebar_position: "8"
 ---
 # Middleware
 
-WIP
\ No newline at end of file
+## What is a "Middleware" and why they are needed
+Before introducing middleware, let us understand why they are needed.
+
+Consider the following example where we have two endpoints within HttpApp
+* GET user by id and
+* GET multiple users paginated
+```scala
+  private val app = Http.collectZIO[Request] {
+    case Method.GET -> !! / "users" / id =>
+      // core business logic  
+      dbService.lookupUsersById(id).map(Response.json(_.json))
+    case Method.GET -> !! / "users"    =>
+      // core business logic  
+      dbService.paginatedUsers(pageNum).map(Response.json(_.json))
+  }
+```
+#### The polluted code violates the principle of "Separation of concerns"
+
+As our application grows, we want to code the following aspects like
+* Basic Auth
+* Request logging
+* Response logging
+* Timeout and retry
+
+For both our example endpoints, our core business logic gets buried under boilerplate like this
```

Comment:
Great example. Can also talk about the concerns of writing unit tests for the core business logic.